test/src/e2e_vm_tests/test_programs/should_fail/bad_generic_var_annotation/test.toml

```diff
@@ -1,13 +1,8 @@
 category = "fail"
 
-# check: let g: str[3] = three_generics(true, "foo", 10);
+# check: $()error
+# check: $()let new_a: B = a;
 # nextln: $()Mismatched types.
-# nextln: $()expected: bool
-# nextln: $()found:    str[3].
-# nextln: $()help: The argument that has been provided to this function's type does not match the declared type of the parameter in the function declaration.
-
-# check: let g: str[3] = three_generics(true, "foo", 10);
-# nextln: $()Mismatched types.
-# nextln: $()expected: str[3]
-# nextln: $()found:    bool.
+# nextln: $()expected: B
+# nextln: $()found:    A.
```

Comment:
This is great, though would it be possible to have both the generic name and the concrete name in the error?

Reply:
I don't think we would want that. The equivalent Rust error for the same code is:

```
error[[E0308]](https://doc.rust-lang.org/stable/error-index.html#E0308): mismatched types
 --> src/main.rs:9:18
  |
6 | fn three_generics<A, B, C>(a: A, b: B, c: C) -> B {
  |                   -  - expected type parameter
  |                   |
  |                   found type parameter
...
9 |   let new_a: B = a;
  |              -   ^ expected type parameter `B`, found type parameter `A`
  |              |
  |              expected due to this
  |
  = note: expected type parameter `B`
             found type parameter `A`
```

See: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a68bc92d1e7dfd75cbee763153d42830